napalm_panos/panos.py

```diff
@@ -31,11 +31,12 @@
 
 from napalm_base.base import NetworkDriver
 from netmiko import ConnectHandler
+from netmiko import __version__ as netmiko_version
 
 
 class PANOSDriver(NetworkDriver):
 
-    def __init__(self, hostname, username, password, timeout=60, optional_args=None):
+    def __init__(self, hostname, username, password='', timeout=60, optional_args=None):
```

Comment:
My vote is to leave this unchanged.

> We could argue that the signature shouldn't have had the username and password to begin with and I might agree but every driver has to have exactly the same method signatures and the same outputs.

I agree with @dbarrosop's statement, we should not make exceptions. If at some point we'll have enough drivers able to use the SSH key instead of the password perhaps we can revisit this.

> Because if you omit the username and the password someone could write code against napalm-panos that will not be compatible with other drivers. 

IMO this sums everything up - it's too much change, not worth it now.